coreneuron/io/reports/report_handler.cpp

```diff
@@ -155,6 +165,77 @@ VarsToReport ReportHandler::get_compartment_vars_to_report(const NrnThread& nt,
     return vars_to_report;
 }
 
+VarsToReport ReportHandler::get_section_vars_to_report(const NrnThread& nt,
+                                                       const std::set<int>& target,
+                                                       double* report_variable,
+                                                       ReportType type) const {
+    VarsToReport vars_to_report;
+    const auto* mapinfo = static_cast<NrnThreadMappingInfo*>(nt.mapping);
+    if (!mapinfo) {
+        std::cerr << "[COMPARTMENTS] Error : mapping information is missing for a Cell group "
+                  << nt.ncell << '\n';
+        nrn_abort(1);
+    }
+
+    for (int i = 0; i < nt.ncell; i++) {
+        int gid = nt.presyns[i].gid_;
+        if (target.find(gid) != target.end()) {
+            CellMapping* cell_mapping = mapinfo->get_cell_mapping(gid);
+            if (cell_mapping == nullptr) {
+                std::cerr
+                    << "[COMPARTMENTS] Error : Compartment mapping information is missing for gid "
+                    << gid << '\n';
+                nrn_abort(1);
+            }
+            std::vector<VarWithMapping> to_report;
+            to_report.reserve(cell_mapping->size());
+
+            std::string name;
+            switch (type) {
+                case AxonReport:
+                case AxonCompReport:
+                    name = "axon";
+                    break;
+                case DendriteReport:
+                case DendriteCompReport:
+                    name = "dend";
+                    break;
+                case ApicalReport:
+                case ApicalCompReport:
+                    name = "apic";
+                    break;
+                default:
+                    name = "soma";
+            }
+            
+            /** get  section list mapping for the type, if available */
+            if (cell_mapping->get_seclist_section_count(name) > 0) {
+                SecMapping* s = cell_mapping->get_seclist_mapping(name);
+                for (auto& sm : s->secmap) {
+                    int compartment_id = sm.first;
+                    auto& vec = sm.second;
+
+                    // Get all the compartment values (otherwise, just middle point)
+                    if (type >= AxonCompReport) {
```

Comment:
semantically this comparison `type >= AxonCompReport` confusing. There is no specific order. Could this be done differently?

Reply:
This is no longer needed with the latest changes, but agreed! It could also introduce errors if someone else changes the order of the `enum` values.